src/itest/kotlin/nl/lifely/zac/itest/SignaleringenRestServiceTest.kt

```diff
@@ -0,0 +1,276 @@
+                response.code shouldBe HttpStatusCode.OK_200.code()
+            }
+        }
+    }
+
+    Given("A zaak with informatie objecten") {
+        val zaakPath = "zaken/api/v1/zaken/$zaak1UUID"
+
+        val zaakInformatieObjectenResponse = itestHttpClient.performGetRequest(
+            url = "$OPEN_ZAAK_EXTERNAL_URI/zaken/api/v1/zaakinformatieobjecten?zaak=$OPEN_ZAAK_EXTERNAL_URI/$zaakPath",
+            openZaakAuth = true
+        )
+        var responseBody = zaakInformatieObjectenResponse.body!!.string()
+        logger.info { "Response: $responseBody" }
+        zaakInformatieObjectenResponse.code shouldBe HttpStatusCode.OK_200.code()
+        val zaakInformatieObjectenUrl = JSONArray(responseBody)
+            .getJSONObject(0)
+            .getString("url")
+        logger.info { "Zaak informatie objecten URL: $zaakInformatieObjectenUrl" }
+
+        val zaakRollenResponse = itestHttpClient.performGetRequest(
+            url = "$OPEN_ZAAK_EXTERNAL_URI/zaken/api/v1/rollen?zaak=$OPEN_ZAAK_EXTERNAL_URI/$zaakPath",
+            openZaakAuth = true
+        )
+        responseBody = zaakRollenResponse.body!!.string()
+        logger.info { "Response: $responseBody" }
+        zaakRollenResponse.code shouldBe HttpStatusCode.OK_200.code()
+        val zaakRollenUrl = JSONObject(responseBody)
+            .getJSONArray("results")
+            .getJSONObject(0)
+            .getString("url")
+            .replace(OPEN_ZAAK_EXTERNAL_URI, OPEN_ZAAK_BASE_URI)
+        logger.info { "Zaak rollen URL: $zaakInformatieObjectenUrl" }
+
+        When("a notification is sent to ZAC that a zaak is updated") {
+            val response = itestHttpClient.performJSONPostRequest(
+                url = "$ZAC_API_URI/notificaties",
+                headers = Headers.headersOf(
+                    "Content-Type",
+                    "application/json",
+                    "Authorization",
+                    ItestConfiguration.OPEN_NOTIFICATIONS_API_SECRET_KEY
+                ),
+                requestBodyAsString = JSONObject(
+                    mapOf(
+                        "actie" to "create",
+                        "kanaal" to "zaken",
+                        "resource" to "zaakinformatieobject",
+                        "hoofdObject" to "$OPEN_ZAAK_BASE_URI/$zaakPath",
+                        "resourceUrl" to zaakInformatieObjectenUrl,
+                        "aanmaakdatum" to ZonedDateTime.now(ZoneId.of("UTC")).toString()
+                    )
+                ).toString(),
+                addAuthorizationHeader = false
+            )
+
+            Then("the response should be 'no content'") {
+                responseBody = response.body!!.string()
+                logger.info { "Response: $responseBody" }
+                response.code shouldBe HttpStatusCode.NO_CONTENT_204.code()
+            }
+        }
+
+        When("a notification is sent to ZAC that a rol is updated") {
```

Comment:
This test should be moved to `NotificationsTest.kt` I think? It has nothing to do with signaleringen right? Or is this a set up for the signaleringen tests later on?

Reply:
Notifications for changes in information object can be sent only after such object exists (file upload).
Assigning the zaak to testuser1 breaks the information object test

So the setup done in the test cannot be easily moved in the other tests

Tests cannot be run in isolation, but I think that's a conscious choice we made - speed over ease of maintenance and clarity? Perhaps we need to rethink that

Reply:
Yes, I agree. Let's think about this some other time. I think it's fine like this.